### PR TITLE
fix(gatsby): open lmdb instances in writeable locations in generated ssr/dsg function

### PR DESCRIPTION
**Pull request by pieh**
## Description

This makes `caches-lmdb` AND `gatsby-core-utils` `getStorage` open lmdb cache/storage instances in writeable location (in case CWD in lambda environment is not writeable). The FS rewrites we have there now doesn't apply to LMDB (fs handling of lmdb is not in Node.js land) so we have to explicitly deal with "fs rewrite" when LMDB is concerned. We were already addressing this for main LMDB instance (datastore), but this expands it to `cache` and `storage` usage of lmdb as well.

`gatsby-core-utils` is already using `global.__GATSBY.root` - https://github.com/gatsbyjs/gatsby/blob/47ec2ac04235d9ae45614a4596a250a44398e9cf/packages/gatsby-core-utils/src/utils/get-storage.ts#L29-L32

This PR adds similar usage to lmdb cache usage, as well as actually sets the root to writeable location in lambda

### Tests

This is example showing segfault problem caused by trying to open LMDB in non-writeable location - https://6527cb3ca4fc76285774d547--exquisite-pavlova-cc785c.netlify.app/

https://6527d2acfad30e0085a57db2--exquisite-pavlova-cc785c.netlify.app/ this one is using changes from this PR showing it fixing the problem.

## Related Issues

Fixes FRA-10
https://github.com/netlify/pillar-support/issues/724

### CODE REVIEW

File: packages/gatsby/src/utils/page-ssr-module/lambda.ts [hunk 1]
```diff
@@ -148,8 +154,8 @@ async function getEngine(): Promise<GraphQLEngineType> {
         reject(error)
       })
     })
+    console.log(`Downloaded datastore from CDN`)
   }
-  console.log(`Downloaded datastore from CDN`)
```
Comment (pieh) [inline packages/gatsby/src/utils/page-ssr-module/lambda.ts:152]: This is unrelated to rest of changes, just noticed this was logged even if LMDB wasn't excluded from the bundle (so when it shouldn't be logged)

**Review by MarcL — APPROVED** [review]
👍🏻

**Comment by madhums** [comment]
@pieh Can we do a release for this fix? Having this issue in production version of code hosted on netlify

**Comment by pieh** [comment]
`gatsby@5.12.7` was released with this change

cc @madhums
